src/client/opamSolution.ml

```diff
@@ -718,8 +718,12 @@ let parallel_apply t ~requested ?add_roots ~assume_built ?(force_remove=false)
           when not (OpamPackage.has_name t.pinned nv.name) ->
           let build_dir =
             OpamPath.Switch.build t.switch_global.root t.switch nv in
-          if not OpamClientConfig.(!r.keep_build_dir) then
-            OpamFilename.rmdir build_dir
+          if not OpamClientConfig.(!r.keep_build_dir) then begin
+            OpamFilename.rmdir build_dir;
+            OpamFilename.remove
```

Comment:
Please define this in `OpamPath` rather than hardcoding the string operations here.

Reply:
Done